### PR TITLE
mk: don't build docs for internal or behind-the-facade crates

### PR DESCRIPTION
**Pull request by emberian**
This saves a bunch of a time and will make distributions smaller, as well as
avoiding filling the implementors page with internal garbage. Turn it back on
with `--enable-compiler-docs` if you want them.

(Crates behind the facade are not documented at all)

[breaking-change]

### CODE REVIEW

**Comment by huonw** [comment]
Doesn't this mean that `core` etc. never have their own explicit documentation?

**Comment by cmr** [comment]
@huonw Indeed not. I'll add a cfg for that (probably keying off whether the channel is stable/beta/nightly)

File: configure [hunk 1]
```diff
@@ -526,7 +526,8 @@ VAL_OPTIONS=""
 opt valgrind 0 "run tests with valgrind (memcheck by default)"
 opt helgrind 0 "run tests with helgrind instead of memcheck"
 opt valgrind-rpass 1 "run rpass-valgrind tests with valgrind"
-opt docs     1 "build documentation"
+opt docs     1 "build standard libraary documentation"
```
Comment (huonw) [inline configure]: Extraa a

File: mk/crates.mk [hunk 2]
```diff
@@ -122,17 +122,25 @@ ONLY_RLIB_rustc_bitflags := 1
...
-              $(filter-out rustc_typeck, \
-              $(filter-out rustc_borrowck, \
-              $(filter-out rustc_resolve, \
-              $(filter-out rustc_driver, \
-              $(filter-out rustc_privacy, \
-              $(filter-out rustc_lint, \
-              $(filter-out log, \
-              $(filter-out getopts, \
-              $(filter-out syntax, $(CRATES))))))))))))
+	$(filter-out rustc_trans, \
```
Comment (huonw) [inline mk/crates.mk]: Unnecessary indentation change?

File: mk/crates.mk [hunk 3]
```diff
@@ -122,17 +122,25 @@ ONLY_RLIB_rustc_bitflags := 1
 # You should not need to edit below this line
 ################################################################################
 
+ifeq ($(CFG_RELEASE_CHANNEL),stable)
```
Comment (huonw) [inline mk/crates.mk]: Could you add a brief comment describing the motivation?

**Comment by huonw** [comment]
r=me with minor nits.

**Comment by steveklabnik** [comment]
:confetti_ball:

**Comment by alexcrichton** [comment]
Hm, was it considered that most `[src]` links will break? The only way for them to work currently is to link back to the code in the original crate, so this will cause it so many of them unfortunately break due to being under the facade.

**Comment by cmr** [comment]
@alexcrichton That is quite unfortunate...

**Comment by steveklabnik** [comment]
Ahhh

**Comment by alexcrichton** [comment]
Actually on second thought this *also* breaks all cross-crate links to `Vec`, for example. Right now a cross-crate link to `Vec` only knows that it was defined in libcollections, not that it was reexported in the standard library.

I think that these two downsides are pretty serious, @cmr would you be ok reverting the "don't build docs for facade crates" part of this PR?

**Comment by cmr** [comment]
@alexcrichton yeah, I'll back that bit out.

**Comment by alexcrichton** [comment]
Ok, thanks @cmr!

**Comment by steveklabnik** [comment]
This broke so many links in documentation: https://github.com/rust-lang/rust/issues/14515#issuecomment-88266890

**Comment by cmr** [comment]
@steveklabnik what docs link to compiler crates?

**Comment by steveklabnik** [comment]
We have an 'unstable' section of the book, and one of the chapters documents syntax extensions.

**Comment by pnkfelix** [comment]
( I think @cmr's aforementioned "back that bit out" commit was this one: b314fedc4a14ac897a0f1dbd3b84130ec82c94c9 )

**Pull request merged**